nats/aio/transport.py

```diff
@@ -139,6 +139,13 @@ async def connect_tls(
         buffer_size: int,
         connect_timeout: int,
     ):
+        # TODO(@orsinium): this assertion is wrong but so the implementation,
+        # and IDK how to fix it.
+        assert self._io_writer
```

Comment:
what is the error reported by mypy? This would have been used here: https://github.com/nats-io/nats.py/blob/48059d5191fc8626e6abc85509c54413319c3585/nats/aio/client.py#L1893-L1898

Reply:
`self._io_writer` is defined as `Optional[StreamWriter]` and is defined in 3 places:

1. In `__init__` as None
2. In `connect` using `asyncio.open_connection`.
3. In `connect_tls` at line 169 but AFTER it is used at line 156.

So, either line 156 fails with AttributeError or `self._io_writer` is defined somewhere else. Is it defined? Is `connect` supposed to be called before `connect_tls`?

Reply:
I think so, first `connect` would happen in `select_next_server` which would establish the TCP connection, then after getting the initial `INFO` it would upgrade with `connect_tls`:
https://github.com/nats-io/nats.py/blob/48059d5191fc8626e6abc85509c54413319c3585/nats/aio/client.py#L1302-L1308

Reply:
Thank you! Fixed: https://github.com/nats-io/nats.py/pull/384/commits/a826001f5a2e13ba34ffc1e158979d232370a1d2